File: app/src/main/java/harkor/mycryptocurrency/view/DialogAdd.java
package harkor.mycryptocurrency.view;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import harkor.mycryptocurrency.Interfaces.AddDialogInterface;
import harkor.mycryptocurrency.Interfaces.CryptoAdd;
import harkor.mycryptocurrency.Interfaces.ListRefresh;
import harkor.mycryptocurrency.Interfaces.ToastDisplay;
import harkor.mycryptocurrency.R;
import harkor.mycryptocurrency.services.CryptoCheckAdd;
import harkor.mycryptocurrency.services.DatabaseController;
import harkor.mycryptocurrency.viewmodel.AddDialogViewmodel;

public class DialogAdd extends DialogFragment implements AddDialogInterface, ToastDisplay{

Context context;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        final AddDialogViewmodel addDialogViewmodel=new AddDialogViewmodel(this);
        AlertDialog.Builder builder=new AlertDialog.Builder(getActivity());
        final View mView=getActivity().getLayoutInflater().inflate(R.layout.add_dialog,null);
        context=mView.getContext();
        builder.setView(mView)
            .setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialogInterface, int i) {
                }
            })
            .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialogInterface, int i) {
                    EditText editName= mView.findViewById(R.id.edit_name);
                    EditText editAmount=mView.findViewById(R.id.edit_amount);
                    addDialogViewmodel.okClick(editName.getText()+"",editAmount.getText()+"");
                }
            });
        return builder.create();

    }

    @Override
    public ListRefresh getListRefresh() {
        return (ListRefresh)getActivity();
    }

    @Override
    public DatabaseController getDatabase() {
        return new DatabaseController(context);
    }

    @Override
    public ToastDisplay getToastDisplay() {
        return this;
    }

    @Override
    public void showToast(int toastId) {
        switch (toastId){
            case 1:
                Toast.makeText(context,R.string.bad_name_amount,Toast.LENGTH_SHORT).show();
                break;
            case 2:
                Toast.makeText(context,R.string.connection_error,Toast.LENGTH_SHORT).show();
                break;
        }
    }
}

File: app/src/main/java/harkor/mycryptocurrency/view/DialogAdd.kt
package harkor.mycryptocurrency.view

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.view.LayoutInflater
import android.widget.EditText
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import harkor.mycryptocurrency.R


class DialogAdd(private val noticeAddDialogListener: NoticeAddDialogListener) : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return activity?.let {
            val builder = AlertDialog.Builder(it)
            val dialogView = LayoutInflater.from(activity).inflate(R.layout.add_dialog, null)
            dialogView.findViewById<TextView>(R.id.add_cancel).setOnClickListener {
                dialog?.cancel()
            }
            dialogView.findViewById<TextView>(R.id.add_ok).setOnClickListener {
                val name = dialogView.findViewById<EditText>(R.id.add_edit_name).text.toString()
                val amount = dialogView.findViewById<EditText>(R.id.add_edit_amount).text.toString().toDoubleOrNull()
                if (amount != null && name != "") {
                    noticeAddDialogListener.addNewCrypto(name, amount)
                    dialog?.cancel()
                }
            }
            builder.setView(dialogView)
            builder.create()
        } ?: throw IllegalAccessException("Activity cannot be null")
    }
}

interface NoticeAddDialogListener {
    fun addNewCrypto(name: String, amount: Double)
}